x/proof/keeper/keeper.go

```diff
@@ -32,18 +43,45 @@ func NewKeeper(
 	authority string,
 
 	sessionKeeper types.SessionKeeper,
+	applicationKeeper types.ApplicationKeeper,
+	accountKeeper types.AccountKeeper,
 ) Keeper {
 	if _, err := sdk.AccAddressFromBech32(authority); err != nil {
 		panic(fmt.Sprintf("invalid authority address: %s", authority))
 	}
 
+	applicationQuerier := types.NewAppKeeperQueryClient(applicationKeeper)
+	accountQuerier := types.NewAccountKeeperQueryClient(accountKeeper)
+	// TODO_TECHDEBT: Use cosmos-sdk based polylog implementation once available. Also remove polyzero import.
+	polylogger := polylog.Ctx(context.TODO())
+
+	ringClientDeps := depinject.Supply(
```

Comment:
Can you add a few comments on what's happening here as we set up the clients and why?

Reply:
@red-0ne Given both mine & @h5law's confusion, this definitely needs a #PUC on what/why is happening.